core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CloudflowAkkaPlugin.scala

```diff
@@ -0,0 +1,129 @@
+import com.typesafe.sbt.packager.Keys._
+import spray.json._
+import cloudflow.sbt.CloudflowKeys._
+import cloudflow.blueprint.StreamletDescriptorFormat._
+import cloudflow.sbt.CloudflowKeys._
+
+object CloudflowAkkaPlugin extends CloudflowBasePlugin {
+
+  override def projectSettings = Seq(
+    libraryDependencies ++= Vector(
+      "com.lightbend.cloudflow" %% "cloudflow-akka-util" % BuildInfo.version,
+      "com.lightbend.cloudflow" %% "cloudflow-akka" % BuildInfo.version,
+      "com.lightbend.cloudflow" % "cloudflow-runner" % BuildInfo.version,
+      "com.lightbend.cloudflow" %% "cloudflow-akka-testkit" % BuildInfo.version % "test"
+    ),
+
+    cloudflowAkkaDockerImageName := Def.task {
+      Some(DockerImageName((ThisProject / name).value.toLowerCase, (ThisProject / cloudflowBuildNumber).value.buildNumber))
+    }.value,
+
+    streamletDescriptorsInProject := Def.taskDyn {
+      val detectedStreamlets = cloudflowStreamletDescriptors.value
+      buildStreamletDescriptors(detectedStreamlets)
+    }.value,
+
+    buildOptions in docker := BuildOptions(
+      cache = true,
+      removeIntermediateContainers = BuildOptions.Remove.OnSuccess,
+      pullBaseImage = BuildOptions.Pull.IfMissing
+    ),
+
+    cloudflowStageAppJars := Def.taskDyn {
+      Def.task {
+        val stagingDir = stage.value
+        val projectJars = (Runtime / internalDependencyAsJars).value.map(_.data)
+        val depJars = (Runtime / externalDependencyClasspath).value.map(_.data)
+
+        val appJarDir = new File(stagingDir, AppJarsDir)
+        val depJarDir = new File(stagingDir, DepJarsDir)
+        projectJars.foreach { jar ⇒
+          IO.copyFile(jar, new File(appJarDir, jar.getName))
+        }
+        depJars.foreach { jar ⇒
+          IO.copyFile(jar, new File(depJarDir, jar.getName))
+        }
+      }
+    }.value,
+
+    imageNames in docker := {
+      val registry = cloudflowDockerRegistry.value
+      val namespace = cloudflowDockerRepository.value.orElse(registry.map(_ ⇒ "lightbend"))
+
+      cloudflowAkkaDockerImageName.value
+        .map { imageName ⇒
+          ImageName(
+            registry = registry,
+            namespace = namespace,
+            repository = imageName.name,
+            tag = Some(imageName.tag)
+          )
+        }
+        .toSeq
+    },
+
+    dockerfile in docker := {
+      // this triggers side-effects, e.g. files being created in the staging area
+      cloudflowStageAppJars.value
+
+      val appDir: File = stage.value
+      val appJarsDir: File = new File(appDir, AppJarsDir)
+      val depJarsDir: File = new File(appDir, DepJarsDir)
+
+      // pack all streamlet-descriptors into a Json array
+      val streamletDescriptorsJson =
+        streamletDescriptorsInProject
+          .value
+          .toJson
+
+      val streamletDescriptorsLabelValue = makeStreamletDescriptorsLabelValue(streamletDescriptorsJson)
+
+      new Dockerfile {
+        from("adoptopenjdk/openjdk8")
```

Comment:
Should this be an individual decision for each plugin, or should we standardize this? My thinking here is that we should always provide the same JVM for all Streamlet types, Akka, Spark, Flink etc. Wonder if the creation of the docker file could be the same for all these plugins?

Reply:
I agree that it would be good to make sure that all three images are based on the same base if that is at all possible.

Reply:
@olofwalker we are using this same jdk across all plugins. For Spark and Flink we have separate scripts that create the base images and all of them use this same jdk. For Akka, we don't need a separate step for creating a base image - hence we have it in the plugin.

Reply:
This could also be part of the top abstract class or `CommonSettingsAndTasksPlugin` to underline the fact we are re-using a property, just in case we support new runtimes inthe future like kafka streams.

Reply:
I think this is fine, like this. (assuming it is not an outstanding issue that needs resolution)